docs/user_guides/fs/tags/tags.md

```diff
@@ -0,0 +1,138 @@
+
+```
+{
+  "type" : "object",
+  "properties" :
+  {
+    "first_name" : { "type" : "string" },
+    "last_name" : { "type" : "string" },
+    "age" : { "type" : "integer" },
+    "hobbies" : {
+        "type" : "array",
+        "items" : { "type" : "string" }
+    }
+  },
+  "required" : ["first_name", "last_name", "age"],
+  "additionalProperties": false
+}
+```
+
+Additionally it is also possible to define single property as tag. This is achieved by defining a JSON schema like the following:
+
+```
+{ "type" : "string" }
+```
+
+Where the type is a valid primitive: `string`, `boolean`, `integer`, `number`.
+
+## Step 2: Attach a tag to an artifact
+
+Once the schema has been created, you can attach a tag with that schema to a feature group, feature view or training dataset either using the feature store APIs, or by using the UI.
+
+### Using the API
+
+You can attach tags to feature groups and feature views by using the `add_tag()` method of the feature store APIs:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Define the tag
+    tag = {
+        'business_unit': 'Fraud',
+        'data_owner': 'email@hopsworks.ai',
+        'pii': True
+    }
+
+    # Attach the tag
+    fg.add_tag("data_privacy", tag) 
+    ```
+
+You can see the list of tags attached to a given artifact by using the `get_tags()` method:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Retrieve the tags for this feature group 
+    fg.get_tags()
+    ```
+
+Finally you can remove a tag from a given artifact by calling the `delete_tag()` method:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Retrieve the tags for this feature group 
+    fg.delete_tag("data_privacy")
+    ```
+
+The same APIs work for feature views and training dataset alike.
+
+### Using the UI
+
+You can attach tags to feature groups and feature views directly from the UI. You can navigate on the artifact page and click on the `Add tags` button. From there you can select the schema of the tag you want to attach and populate the values as shown in the gif below.
+
+<p align="center">
+  <figure>
+    <a  href="../../../../assets/images/guides/tags/tags_ui.gif">
+      <img src="../../../../assets/images/guides/tags/tags_ui.gif" alt="Attach tag to a feature group">
+    </a>
+    <figcaption>Attach tag to a feature group</figcaption>
+  </figure>
+</p>
+
+## Step 3: Search
+
+Hopsworks indexes the tags attached to feature groups, feature views and training dataset. The tags will then be searchable using the free text search feature available in the UI.
```

Comment:
, and training datasets

the free text search box located at the top of the UI.